src/datasets/arrow_dataset.py

```diff
@@ -707,6 +707,40 @@ def unique(self, column: str) -> List[Any]:
 
         return self._data.column(column).unique().to_pylist()
 
```

Comment:
@lhoestq @SBrandeis `class_encode_column` modifies the features of the dataset and that means this method should be decorated with the `update_metadata_with_features` decorator.

```suggestion
@update_metadata_with_features
```